shell.nix

```diff
@@ -0,0 +1,18 @@
+{ pkgs ? import <nixpkgs> {} }:
+with pkgs;
+
+let
+in pkgs.callPackage ./derivation.nix {
+  buildGoModule = super: buildGoModule (super // {
```

Comment:
As I am quite new in programming language and nix in general, I might need a little extra help to understand these two lines 5 and 6! So of what I understand, `buildGoModule` is a function that should return a set that will represent the final derivation. So defining it like this woud allow the caller to provide a `super` set argument that would be merged with the body of the function here.

But what is the difference with having maybe something a little bit simpler like the following:

**derivation.nix**
```
{ stdend, overrides }:

overrides // {
    name = "rMAPI";
    src = ./.;
    goDeps = ./deps.nix
    ...
}
```

**shell.nix**
```
{ pkgs ? import <nixpkgs> {} }:

pkgs.callPackage ./derivation.nix {
    overrides = {
        name = "rMAPI-env";
        buildInputs = ...;
    };
}
```

Thanks for shedding light!

Reply:
I do think, technically this would work. But I can see some possible downsides of this approach:
* I tried to to make the inputs reflect dependencies only 
* callPackage will resolve `overrides` to whatever this is defined to in the scope of `<nixpkgs>`
* [overriding describes the process of changing parameters of a function](https://nixos.org/nixos/nix-pills/nixpkgs-overriding-packages.html) and therefore `overrides` is not a good name for this. See [this discussion](https://discourse.nixos.org/t/best-practices-for-config-flags-to-packages/4064) for some ideas around the topic.

Also I liked the readability and simplicity of the chosen approach. But that's just my opinion, maybe i am completely wrong about this.